boa_engine/src/vm/call_frame/abrupt_record.rs

```diff
@@ -0,0 +1,77 @@
+//! Implements an `AbruptCompletionRecord` struct for `CallFrame` tracking.
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub(crate) enum AbruptKind {
+    None,
+    Continue,
+    Break,
+    Throw,
+}
```

Comment:
Not really an exact reason. It was more to provide a value that could be used during a default initialization.

Reply:
Hm, then I think it would be better to remove the default implementation of `AbruptCompletionRecord` or make the completion type in `Codeblock` a pure `AbruptCompletionRecord`. I'm tending towards the first option just because it doesn't make much sense to have an abrupt completion record with kind `None`, but if the implementation doesn't allow it then the second option is alright.

Reply:
Also, I think it would be better if `AbruptCompletionRecord` had a single constructor that received the target and the kind, instead of having a construction for every kind.

Reply:
Would it be better to just use some creation methods like `create_break_completion`, `create_throw_completion`, and `create_continue_creation`. I was trying to avoid having to import `AbruptKind` in multiple places, and making that change might also make the creation more clear/readable.

Reply:
I could get behind that! We can remove the `Default` impl that way